jax/experimental/compilation_cache/gfile_cache.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+
+from jax.experimental.compilation_cache.cache_interface import CacheInterface
+from etils import epath
+from absl import logging
+
+class GFileCache(CacheInterface):
+
+  def __init__(self, path: str):
+    """Sets up a cache at 'path'. Cached values may already be present."""
+    self._path = epath.Path(path)
+    self._path.mkdir(parents=True, exist_ok=True)
+
+  def get(self, key: str):
+    """Returns None if 'key' isn't present."""
+    if not key:
+      raise ValueError("key cannot be empty")
+    path_to_key = self._path / key
+    if path_to_key.exists():
+      return path_to_key.read_bytes()
+    else:
+      return None
+
+  def put(self, key: str, value: bytes):
+    """Adds new cache entry."""
+    if not key:
+      raise ValueError("key cannot be empty")
+    path_to_new_file = self._path / key
+    if 'gs://' in str(path_to_new_file):
+      path_to_new_file.write_bytes(value)
+    else:
+      tmp_path = epath.Path(str(self._path / "_") + key)
```

Comment:
Use f-strings here instead of `+`?

Reply:
Also maybe you don't need another epath use?

`tmp_path = self._path / f"_temp_{key}"`?

Reply:
Can you try using this please? What you have right now is very confusing.

Reply:
aaah, I see, I didn't see your second comment! :D yeah sure.